### PR TITLE
build: cmake: Use portable delimiters.

### LINKED ISSUE #5342: Generating nvim.pot
os           :  win7-64 
platform : msys2 mingw64

CMake Error at F:/CProject/neovim/cmake/RunXgettext.cmake:4 (list):
  list sub-command SORT requires list to be present.
src\nvim\po\CMakeFiles\translations.dir\build.make:378: recipe for target 'src/nvim/po/nvim.pot' failed
mingw32-make.exe[2]: **\* [src/nvim/po/nvim.pot] Error 1
CMakeFiles\Makefile2:1230: recipe for target 'src/nvim/po/CMakeFiles/translations.dir/all' failed
mingw32-make.exe[1]: **\* [src/nvim/po/CMakeFiles/translations.dir/all] Error 2
Makefile:148: recipe for target 'all' failed
mingw32-make.exe: **\* [all] Error 2

I do not know what the problem is not dependent loss.

### PR DESCRIPTION
**Pull request by justinmk**
Fixes build on Windows (msys2 mingw64).
Closes #5342

Result of `build/src/nvim/po/nvim.pot` is identical after this change.

cc @jszakmeister

### CODE REVIEW

**Comment by jszakmeister** [comment]
LGTM.

File: src/nvim/po/CMakeLists.txt [hunk 1]
```diff
@@ -48,7 +48,7 @@ if(HAVE_WORKING_LIBINTL AND GETTEXT_FOUND AND XGETTEXT_PRG AND ICONV_PRG)
         -DXGETTEXT_PRG=${XGETTEXT_PRG}
         -DPOT_FILE=${NVIM_POT}
         -DSEARCH_DIR=${CMAKE_CURRENT_SOURCE_DIR}
-        "'-DSOURCES=${NEOVIM_RELATIVE_SOURCES}'"
+        "\"-DSOURCES=${NEOVIM_RELATIVE_SOURCES}\""
```
Comment (jszakmeister) [inline src/nvim/po/CMakeLists.txt]: Actually, I think you can get rid of the escaped quotes all together.  CMake's expansion rules are different than the shell, and IIRC, the whole thing will get passed as a single argument.

Comment (justinmk) [inline src/nvim/po/CMakeLists.txt]: @jszakmeister I tried that but `build/src/nvim/po/nvim.pot` didn't generate correctly. My assumption was that you were intending to send the literal string

    "-DSOURCES=${NEOVIM_RELATIVE_SOURCES}"

for some reason?

**Comment by justinmk** [comment]
@jszakmeister Thanks for the sanity check!

**Comment by jszakmeister** [comment]
You're welcome!

**Pull request merged**